components/schemes/Color/src/MDCSemanticColorScheme.h

```diff
@@ -0,0 +1,144 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+#import "MDCColorScheme.h"
+
+/**
+ A simple color scheme that provides semantic context for the colors it uses. There are no optional
+ properties and all colors must be provided, supporting more reliable color theming.
+ */
+@protocol MDCColorScheming
+
+/**
+ Displayed most frequently across your app.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColor;
+
+/**
+ A tonal variation of primary color.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *primaryColorVariant;
+
+/**
+ Accents select parts of your UI.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *secondaryColor;
+
+/**
+ The color used to indicate error status.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *errorColor;
+
+/**
+ The color of surfaces such as cards, sheets, menus.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *surfaceColor;
+
+/**
+ The underlying color of an app’s content.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *backgroundColor;
+
+/**
+ A color that passes accessibility guidelines for text/iconography when drawn on top of
+ @c primaryColor.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *onPrimaryColor;
+
+/**
+ A color that passes accessibility guidelines for text/iconography when drawn on top of
+ @c secondaryColor.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *onSecondaryColor;
+
+/**
+ A color that passes accessibility guidelines for text/iconography when drawn on top of
+ @c surfaceColor.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *onSurfaceColor;
+
+/**
+ A color that passes accessibility guidelines for text/iconography when drawn on top of
+ @c backgroundColor.
+ */
+@property(nonnull, readonly, nonatomic) UIColor *onBackgroundColor;
+@end
```

Comment:
onError?
go/m2-color-overview

Reply:
Again, not in pivotal, so I'll let the doc authors hash-out what's going to be built.

Reply:
We'll add these parameters in a follow-up pass once the parameter mappings are more clearly defined.